### PR TITLE
Update the training guide

### LINKED ISSUE #232: Improve model training guide
Reorganize [the doc](https://github.com/mozilla/firefox-translations-training/blob/main/docs/training-guide.md) and add step-by-step instructions for training configuration for easier onboarding.

### PR DESCRIPTION
**Pull request by eu9ene**
The goal is to make it easier to onboard and start training models.

Also deployed to the Github pages: https://mozilla.github.io/firefox-translations-training/

Fixes #232 

[skip ci]

### CODE REVIEW

File: docs/training-guide.md [hunk 1]
```diff
@@ -1,17 +1,177 @@
...
+
+First, choose a language pair to train.
+
+Considerations:
+- The size of the parallel corpus on [OPUS](https://opus.nlpl.eu/)
+- Availability of monolingual data. The pipeline requires monolingual data in both source and target languages. 
+  Currently we support automatic donwloading only for [news crawl](https://data.statmt.org/news-crawl/)
+- Availability of [bicleaner-ai models](https://github.com/bitextor/bicleaner-ai-data/releases)
+
+Set the language pair and a name of the experiment in the config:
```
Comment (marco-c) [inline docs/training-guide.md:26]: Just by reading this, it isn't clear what the "config" is

File: docs/training-guide.md [hunk 2]
```diff
@@ -1,17 +1,177 @@
...
+- Noisier datasets like OpenSubtitles should have higher threshold. 
+- Set the threshold to `0` to skip cleaning entirely, for example for ParaCrawl dataset that comes already cleaned.
+
+```
+  bicleaner:
+    default-threshold: 0.5
+    dataset-thresholds:
+      opus_CCAligned/v1: 0.7
+      opus_OpenSubtitles/v2018: 0.8
+      opus_ParaCrawl/v8: 0
```
Comment (marco-c) [inline docs/training-guide.md:145]: nit: Let's use v9 since it's the latest (I know it's just an example)

File: docs/training-guide.md [hunk 3]
```diff
@@ -1,17 +1,177 @@
...
+For more advanced cleaning and for using OpusCleaner look at the [Data cleaning](cleaning.md) doc.
+
+### Bicleaner
+It is recommended to use Bicleaner ML models to filter noisy data. 
+Check that the bicleaner-ai model is [available](https://github.com/bitextor/bicleaner-ai-data/releases) 
+and add filtering thresholds to the config. 
+
+- `0.5` should be a good default value.
+- Noisier datasets like OpenSubtitles should have higher threshold. 
+- Set the threshold to `0` to skip cleaning entirely, for example for ParaCrawl dataset that comes already cleaned.
```
Comment (marco-c) [inline docs/training-guide.md:137]:
```suggestion
- Set the threshold to `0` to skip cleaning entirely, for example for ParaCrawl dataset that comes already cleaned by bicleaner.
```

And we could add a reference to https://aclanthology.org/2022.lrec-1.87.pdf, section 4.2.2.

File: docs/training-guide.md [hunk 4]
```diff
@@ -1,17 +1,177 @@
...
+Make sure the language is present in [clean_parallel](https://github.com/mozilla/firefox-translations-training/tree/main/pipeline/clean/tools/clean_parallel.py#L19) script.
+
+For more advanced cleaning and for using OpusCleaner look at the [Data cleaning](cleaning.md) doc.
+
+### Bicleaner
+It is recommended to use Bicleaner ML models to filter noisy data. 
+Check that the bicleaner-ai model is [available](https://github.com/bitextor/bicleaner-ai-data/releases) 
+and add filtering thresholds to the config. 
+
+- `0.5` should be a good default value.
```
Comment (marco-c) [inline docs/training-guide.md:135]: I couldn't find this referenced anywhere in the bicleaner-ai repo, where did you find it?

File: docs/cleaning.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+- `<dataset_name>.sh` for parallel dataset cleaning
+- `<dataset_name>.<lang>.sh` for language specific cleaning of parallel or monolingual dataset
+- `/` in dataset name should be replaced with `_`
+
+### Cleaning scripts
+
+Make sure the language is present in [clean_parallel](https://github.com/mozilla/firefox-translations-training/tree/main/pipeline/clean/tools/clean_parallel.py#L19) script.
+
+
+### Bicleaner
```
Comment (marco-c) [inline docs/cleaning.md:33]: This is repeated in the training guide, it'd be better to have a single instance of it so we don't risk forgetting to update one or the other

**Review by gregtatum — APPROVED** [review]
Praise: Great docs! Thanks for taking the time to write them up. I should try and use them soon ;)

File: docs/training-guide.md (hunk 1 above)
Comment (eu9ene) [reply docs/training-guide.md]: updated

File: docs/training-guide.md (hunk 2 above)
Comment (eu9ene) [reply docs/training-guide.md]: fixed

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+layout: default
+title: Home
+nav_order: 1
+description: "Firefox Translations Training documentation."
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/index.md:5]: In this case, FTT is the name of the repo. Also, this is a meta description, so not visible.

File: docs/training-guide.md (hunk 4 above)
Comment (eu9ene) [reply docs/training-guide.md]: added link

File: docs/data.md [hunk 7]
```diff
@@ -25,7 +27,7 @@ Custom parallel | custom-corpus | /tmp/test-corpus | corpus | Custom parallel da
 [Common crawl](https://commoncrawl.org/) | commoncrawl | wmt16 | mono | Huge web crawl datasets. The links are posted on [WMT21](https://www.statmt.org/wmt21/translation-task.html)
 Custom mono | custom-mono | /tmp/test-mono | mono | Custom monolingual dataset that is already downloaded to a local disk. The dataset name is an absolute path prefix without ".lang.gz"
 
-You can also use [find-corpus](/pipeline/utils/find-corpus.py) tool to find all datasets for an importer and get them formatted to use in config.
+You can also use [find-corpus](https://github.com/mozilla/firefox-translations-traininghttps://github.com/mozilla/firefox-translations-training/tree/main/pipeline/utils/find-corpus.py) tool to find all datasets for an importer and get them formatted to use in config.
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/data.md]: fixed

File: docs/training-guide.md [hunk 8]
```diff
@@ -1,17 +1,177 @@
...
+Copy the datasets in the training config:
+```
+datasets:
+  train:
+    - opus_ada83/v1
+    - mtdata_Statmt-news_commentary-15-eng-rus
+    ...
+```
+It's hard to say how much data is required to train something useful. 
+My guess would be at least 10 million sentences. Ideally 100M+.
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/training-guide.md]: removed

File: docs/training-guide.md [hunk 9]
```diff
@@ -1,17 +1,177 @@
...
+    - sacrebleu_wmt17
+  # datasets for evaluation
+  test:
+    - flores_devtest
+    - sacrebleu_wmt20
+    - sacrebleu_wmt18
+```
+
+### Monolingual corpus
+It is recommended to always use back-translations to augment training data and to use
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/training-guide.md]: I expanded on those two here and added links to papers

File: docs/training-guide.md [hunk 10]
```diff
@@ -1,17 +1,177 @@
...
+  use-opuscleaner: false
+```
+Make sure the language is present in [clean_parallel](https://github.com/mozilla/firefox-translations-training/tree/main/pipeline/clean/tools/clean_parallel.py#L19) script.
+
+For more advanced cleaning and for using OpusCleaner look at the [Data cleaning](cleaning.md) doc.
+
+### Bicleaner
+It is recommended to use Bicleaner ML models to filter noisy data. 
+Check that the bicleaner-ai model is [available](https://github.com/bitextor/bicleaner-ai-data/releases) 
+and add filtering thresholds to the config.
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/training-guide.md]: Added

File: docs/training-guide.md [hunk 11]
```diff
@@ -1,17 +1,177 @@
...
+    dataset-thresholds:
+      opus_CCAligned/v1: 0.7
+      opus_OpenSubtitles/v2018: 0.8
+      opus_ParaCrawl/v8: 0
+      ...
+```
+
+## 4. Set hyperparameters
+
+The pipeline supports overriding the default [Marian settings](https://marian-nmt.github.io/docs/cmd/marian/) in the training config.
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/training-guide.md:163]: They are pretty huge, I added links to the default configs

File: docs/orchestrators.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+nav_order: 6
+has_children: true
+has_toc: false
+---
+
+# Orchestrators
+
+An orchestrator is responsible for workflow management and parallelization.
+
+- [Taskcluster](https://taskcluster.net/) - Mozilla task execution framework. It is also used for Firefox CI.
```
[the remark this answers is not in the document]
Comment (eu9ene) [reply docs/orchestrators.md:15]: It was already mentioned but I made it more visible

File: docs/cleaning.md (hunk 5 above)
Comment (eu9ene) [reply docs/cleaning.md:33]: Good point, I added it for completeness. Replaced with a reference to the section in the training guide now. It's important to have it there I think.

**Review by marco-c — APPROVED** [review]
Thanks! I haven't tried training a model yet, but this is a very good start.

**Pull request merged**